### DISCUSSION TITLE
Access to Roast Properties data programmatically

### DISCUSSION THREAD

**Discussion by jsphvrghs** [artisan-roaster-scope/artisan#1326] [General]
Hello.
I am doing some data analysis and would like to avoid having to manually enter data in my spreadsheet. 
Is there any way to access data from the Roast Properties Panel programmatically through a Python script for eg. 

![image](https://github.com/artisan-roaster-scope/artisan/assets/125485444/f6009373-7717-49e5-b5d1-0fee9277167a)

Thank you.

(P.S. Image source: https://artisan-scope.org/docs/properties/)

**Comment by MAKOMO** [comment]
No, but all data is stored in the .alog profiles which can also be exported as JSON. You should be able to put it from there into an Excel format with a small script easily.

**Reply by MAKOMO** [reply]
Some of the data is also available in Ranking Reports which can be saved in Excel format directly. Just choose File >> Roast >> Report >> Ranking >> Excel (or CSV), choose a bunch of .alog roast profiles and save the file.

If this does not fit, fork the code and modify it to your needs.

**Reply by jsphvrghs** [reply]
Thanks... that seems like a workable solution.

**Reply by jsphvrghs** [reply]
I thought of treating the alog file as a JSON data. However I ran into two small issues :
1)JSON expects double quotes but the alog file uses single quotes
2)JSON expects false and true, but the alog file has False and True

Since I'm new to Python and JSON and only learning as I go, it took me a while to figure this. (The error messages from the json parsers in Python are helpful onlty to a certain extent). So, I was wondering if the alog file writes it the current way due to some design constraint.

**Reply by MAKOMO** [reply]
Artisan files are not in JSON format. Use menu File >> Export >> Artisan JSON or File >> Convert To >> Artisan JSON.

**Reply by jsphvrghs** [reply]
Will keep that in mind.... For now I'm all set with the two minor on-the-fly modifications to the alog file after reading it. 
Many thanks for the pointers.

**Comment by CarefreeBuzzBuzz** [comment]
Are you asking about during a roast or after?

After all data can be downloaded through a Roast Ranking Report.  You can choose multiple roasts to have multiple rows.  
I use this to create custom Roast cards that I put on my dosing rack.

There is a column for each piece of data that you can then use as you like.  

￼
----------------------------------------
Michael Herbert, Head Roaster
Order at www.carefreebuzzbuzz.com <http://www.carefreebuzzbuzz.com/>
Support Artisan RoasterScope  <https://artisan-scope.org/>by using 
Artisan.Plus <https://artisan.plus/en/>

**Reply by jsphvrghs** [reply]
During would be nice ... but looks like that is not possible .. or is it?
I'll explore your suggestion ... sounds like something that will do the work.
Thanks a lot.
